core/crypto/randomness/impl/chain_randomness_provider_impl.cpp

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "crypto/randomness/impl/chain_randomness_provider_impl.hpp"
+
+#include <boost/assert.hpp>
+#include <libp2p/crypto/sha/sha256.hpp>
+#include "common/le_encoder.hpp"
+#include "primitives/ticket/ticket.hpp"
+#include "primitives/tipset/tipset_key.hpp"
+#include "storage/chain/chain_store.hpp"
+
+namespace fc::crypto::randomness {
+
+  using crypto::randomness::Randomness;
+  using primitives::ticket::Ticket;
+  using primitives::tipset::TipsetKey;
+
+  namespace {
+    /**
+     * @brief ChainStore needs its own randomnes calculation function
+     * @return randomness value
+     */
+    crypto::randomness::Randomness drawRandomness(const Ticket &ticket,
+                                                  uint64_t round) {
+      common::Buffer buffer{};
+      const size_t bytes_required = sizeof(round) + ticket.bytes.size();
+      buffer.reserve(bytes_required);
```

Comment:
Why not `buffer.reserve(sizeof(round) + ticket.bytes.size())`?

Reply:
for readability